worlds/mm-recomp/Rules.py

```diff
@@ -296,14 +330,191 @@ def get_location_rules(player):
+        "Great Bay Temple Wart Chest":
+            lambda state.has("Progressive Bow", player),
+        "Great Bay Temple Gekko Entrance Room Caged Chest":
+            lambda state: can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player),
+        "Great Bay Temple Red-Green Pipe First Room Chest":
+            lambda state: can_use_ice_arrows(state, player),
+        "Great Bay Temple Green-Yellow Pipe Chest":
+            lambda state: can_use_ice_arrows(player, state),
+        "Great Bay Temple Green Pipe Freezable Waterwheel Upper Chest":
+            lambda state: can_use_ice_arrows(state, player),
+        "Great Bay Temple Green Pipe Freezable Waterwheel Lower Chest":
+            lambda state: can_use_ice_arrows(state, player),
+        "Great Bay Temple Green Pipe Lever Room Chest":
+            lambda state: can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player),
+        "Great Bay Temple Heart Container":
+            lambda state: (can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player) and state.has("Small Key (Great Bay Temple)", player) and state.has("Boss Key (Great Bay Temple)", player)) or (state.has("Gyorg's Remains", player) and state.has("Progressive Bow", player)),
+        "Great Bay Temple Gyorg's Remains":
+            lambda state: (can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player) and state.has("Small Key (Great Bay Temple)", player) and state.has("Boss Key (Great Bay Temple)", player)) or (state.has("Gyorg's Remains", player) and state.has("Progressive Bow", player)),
+        
 
         "Road to Ikana Pillar Chest":
             lambda state: state.has("Hookshot", player),
         "Road to Ikana Rock Grotto Chest":
             lambda state: state.has("Goron Mask", player),
         "Road to Ikana Stone Soldier":
             lambda state: can_play_song("Epona's Song", state, player) and has_bottle(state, player) and state.has("Progressive Magic Upgrade", player) and state.has("Lens of Truth", player),
+            
+            
+        "Graveyard Day 1 Chest":
+            lambda state: state.has("Captain's Hat", player) and can_smack_hard(state, player) and can_use_fire_arrows(state, player),
+        "Graveyard Day 2 Chest":    
+            lambda state: state.has("Captain's Hat", player) and can_smack_hard(state, player),
+        "Graveyard Day 3 Chest":
+            lambda state: state.has("Captain's Hat", player) and (can_smack_hard(state, player) or has_projectiles(state, player)),
+        "Graveyard Skull Keeta Chest":
+            lambda state: can_play_song("Sonata of Awakening", state, player) and can_smack_hard(state, player),
+        "Song of Storms":    
+            lambda state: state.has("Captain's Hat", player) and can_smack_hard(state, player),
+            
+        "Ikana Canyon Pamela's Father":
+            lambda state: can_play_song("Song of Healing", state, player),
```

Comment:
also needs Song of Storms, to get rid of the Gibdos around the Muic Box, AND either Bombs (or Blast Mask) or Stone Mask:
- **Stone Mask** allowing you to bypass Pamela.
- **Bombs** allowing you to force her out of the house, so you can sneak in.

Reply:
Rule modified to `can_play_song("Song of Healing", state, player) and can_play_song("Song of Storms", state, player) and (has_explosives(state, player) or state.has("Stone Mask", player,)),`